pkg/crossplane/application.go

```diff
@@ -0,0 +1,79 @@
+package crossplane
+
+import (
+	"k8s.io/apimachinery/pkg/api/meta"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+var (
+	fieldsAppDesiredRes          = append(fieldsStatus, "desiredResources")
+	fieldsAppSubmittedRes        = append(fieldsStatus, "submittedResources")
+	fieldsAppResourceSelector    = append(fieldsSpec, "resourceSelector")
+	fieldsAppResourceMatchLabels = append(fieldsAppResourceSelector, "matchLabels")
+
+	stateAppSubmitted = "Submitted"
+)
+
+type Application struct {
+	u *unstructured.Unstructured
+}
+
+func NewApplication(u *unstructured.Unstructured) *Application {
+	return &Application{u: u}
+}
+
+func (o *Application) GetStatus() string {
+	return getNestedString(o.u.Object, fieldsStatusState...)
```

Comment:
I usually prefer that the imports for shared functionality are explicit so they're easier to follow for readers who are not familiar with the codebase, but this is something we can probably reexamine in the future.

Reply:
Well, if you mean `getNestedString` this is not coming from an import. it is a helper in the same package.
Otherwise, I agree with your comment in general.